src/wp-includes/class-wp-query.php

```diff
@@ -3638,6 +3646,8 @@ public function the_post() {
 		$this->in_the_loop = true;
 
 		if ( -1 == $this->current_post ) { // Loop has just started.
+			$this->before_loop = false;
```

Comment:
I wonder if we should move this up to where `$in_the_loop` is set to `true`. That way we keep the assignment unconditional in one place.

Reply:
Yes it did make sense to keep it outside the condition. I have updated the code.